tslearn/early_classification.py

```diff
@@ -0,0 +1,312 @@
+from sklearn.metrics import confusion_matrix
+from sklearn.base import BaseEstimator, ClassifierMixin, clone
+from sklearn.model_selection import train_test_split
+from tslearn.utils import to_time_series_dataset
+from tslearn.clustering import TimeSeriesKMeans
+import numpy as np
+from scipy.sparse import coo_matrix
+
+
+class NonMyopicEarlyClassification(BaseEstimator, ClassifierMixin):
+    """Early Classification modelling for time series using the model presented in [1]
+
+    Parameters
+    ----------
+    n_clusters : int
+        Number of clusters to form.
+
+    base_classifier : Estimator
+        Estimator (instance) to be cloned and used for classifications.
+
+    minimum_time_stamp : int
+        Earliest time at which a classification can be performed on a time series
+
+    lamb : float
+        Value of the hyper parameter lambda used during the computation of the cost function to evaluate the probability
+        that a time series belongs to a cluster given the time series.
+
+    cost_time_parameter : float
+        Parameter of the cost function of time. This function is of the form : f(time) = time * cost_time_parameter
+
+    Attributes
+    --------------------
+    silhouette_ : float
+        The silhouette score from the clustering
+
+    classifiers_ : list
+        A list containing all the classifiers trained for the model
+
+    clusters_ : dictionary
+        Contains the times series by clusters
+
+    pyhatyck_ : dictionary
+        Contains the probabilities of being classified as class y_hat given class y and cluster ck
+
+    indice_ck_ : list
+        Contains for each clusters the indexes of the time series in the dataset
+
+    __n_classes_ : int private
+        The number of classes in the series
+
+    __len_X_ : int
+        The length of the time series
+
+    Examples
+    --------------------
+
+    References
+    --------------------
+    [1] A. Dachraoui, A. Bondu & A. Cornuéjols. Early classification of time series as a non myopic sequential decision
```

Comment:
Please do not use characters with accents (write Cornuejols for once) in python files.